ztp/siteconfig-generator/README.md

```diff
@@ -0,0 +1,29 @@
+## SiteConfig generator
+
+In order to provision an OpenShift cluster using ACM, end user need to define the following CRs (Custom Resources);
+  - AgentClusterInstall
+  - ClusterDeployment
+  - NMStateConfig
+  - KlusterletAddonConfig
+  - ManagedCluster
+  - InfraEnv
+  - BareMetalHost
+  - ConfigMap for extra-manifest configurations
```

Comment:
Saying that the end user needs to create these makes it sound like that is what we are proposing they need to do. Maybe reword to say:

> The siteconfig-generator library makes cluster deployment easier by generating the following CRs for a user based on a SiteConfig CR instance